### PR TITLE
Update code style, partially use new qunit interface, give test module a new name

### PR DESCRIPTION
**Pull request by markelog**
Fixes gh-2056, gh-2437, gh-2540, gh-2571

On gh-2540
See http://qunitjs.com/upgrade-guide-2.x/

For most of the boring work was used https://github.com/apsdehal/qunit-migrate package

However, it can't update local `qunit` helpers, plus in some places deprecated `QUnit.asyncTest` method is still used.

--

Some of these changes strongly depend on each other, others are also fixed, since i touched relevant files, so i decided to put them in one pull, i hope that is okay.

### CODE REVIEW

**Comment by mzgol** [comment]
I'm against the "cache node_modules" part, see https://github.com/jquery/jquery/issues/2541#issuecomment-131523350.

File: src/css/var/rmargin.js [hunk 1]
```diff
@@ -1,3 +1,3 @@
-define(function() {
-	return (/^margin/);
-});
+define( function() {
+	return ( /^margin/ );
```
Comment (mzgol) [inline src/css/var/rmargin.js]: Are the parens still needed? I think it's no longer required by JSHint.

File: test/data/testinit.js [hunk 2]
```diff
@@ -160,7 +165,9 @@ this.ajaxTest = function( title, expect, options ) {
 					if ( options.teardown ) {
 						options.teardown();
 					}
-					start();
+
+					// Make sure all events will be called before done()
+					setTimeout( done );
```
Comment (mzgol) [inline test/data/testinit.js]: This seems dangerous; could you explain why a single `setTimeout` is guaranteed to fix it?

**Comment by mzgol** [comment]
Overall, a good job!

File: test/unit/ready.js [hunk 3]
```diff
@@ -1,18 +1,18 @@
-module( "event" );
+QUnit.module( "ready" );
```
Comment (mzgol) [inline test/unit/ready.js]: How does the dropdown from the screenshot at #2437 look like now? Is the double `event` entry changed to `event` & `ready`? Does choosing `ready` work?

File: test/data/testinit.js (hunk 2 above)
Comment (markelog) [inline test/data/testinit.js]: This is done in case test body listens to ajax events, those events that do fire after `complete` should fire right after it

File: src/css/var/rmargin.js (hunk 1 above)
Comment (markelog) [inline src/css/var/rmargin.js]: Don't know, but that is not a subject of this PR, too much already is squashed in there :-)

File: test/unit/ready.js (hunk 3 above)
Comment (markelog) [inline test/unit/ready.js]: yes and yes

File: build/tasks/build.js [hunk 4]
```diff
@@ -82,13 +89,15 @@ module.exports = function( grunt ) {
 			contents = contents
 				.replace( /define\(\[[^\]]*\]\)[\W\n]+$/, "" );
 		}
+
 		// AMD Name
-		if ( (amdName = grunt.option( "amd" )) != null && /^exports\/amd$/.test( name ) ) {
-			if (amdName) {
+		if ( ( amdName = grunt.option( "amd" ) ) != null && /^exports\/amd$/.test( name ) ) {
```
Comment (gibson042) [inline build/tasks/build.js]: Can changes like this wait for [*SpacesInsideParenthesizedExpression](https://github.com/jscs-dev/node-jscs/pull/1254)?

File: src/css/var/rmargin.js (hunk 1 above)
Comment (mzgol) [inline src/css/var/rmargin.js]: Fair enough. :)

File: build/tasks/build.js (hunk 4 above)
Comment (markelog) [inline build/tasks/build.js]: The rule itself is not such a big thing, **when** (and again, thousands apologies i still didn't get around to that request) this rule will be landed, we would need to update our code style, meaning, we would need to argue and probably vote again, after that, we would need to update documentation on the site, after that we would need to add it to the `jquery` preset and after that, we would need to wait on next the minor release of jscs.

That might take some time, whereas we have our code-style in current form, on which we all agreed upon. Whereas, if we would decide on that, we could auto-format it in the same easy way i did in this pull.

Comment (timmywil) [inline build/tasks/build.js]: > Whereas, if we would decide on that, we could auto-format it in the same easy way i did in this pull.

That's a good point. I don't think we need to wait for that in order to land this PR.

Comment (gibson042) [inline build/tasks/build.js]: I wasn't advocating blocking the PR, just backing out the changes pertaining to spaces in parenthesized expressions (which I believe would shrink it considerably, but I suppose would require a temporary `requireSpacesInsideParentheses` override).

As for revoting/updating documentation/etc., a rule requiring such spaces doesn't appear in http://contribute.jquery.org/style-guide/js/#spacing or the old [.jscs.json](https://github.com/markelog/jquery/commit/b7283d738ec4c2d9d87cb16c32a3209cab567ed1#diff-ed6a67a86c4c5aa1dd5fcb0acb6f5c67), and in fact was stealth-added (from the perspective of this codebase, not its contributors) by https://github.com/jscs-dev/node-jscs/commit/0f78baf08fba75aeba4575608aaa08c384875701 in response to https://github.com/jquery/contribute.jquery.org/pull/104 (which was about requiring spaces in function calls). Since you committed it, I can just directly ask how you would have treated _expression_ parentheses if JSCS were capable of differentiating them. But I will point out that I haven't encountered one argument in favor of requiring them that didn't hinge upon style guide automation, which falls apart as soon as that process becomes sufficiently sophisticated.

Comment (markelog) [inline build/tasks/build.js]: > which was about requiring spaces in function calls

Well, in that discussion Mike suggested to remove all exceptions, so we wouldn't create sophisticated logic to the parentheses rules in JSCS needed to the one preset, that was somewhat the point of that proposal.

> I can just directly ask how you would have treated expression parentheses if JSCS were capable of differentiating them

I would follow the style guide, no matter of it is, i.e. `jquery` preset should represent it.

Personally, i would add more exceptions to that rule as i suggested in jquery/contribute.jquery.org#104  including the "expression parentheses"

Comment (timmywil) [inline build/tasks/build.js]: I'm less bothered by the extra spaces in expression parenthesis as I am by the extra space at the end of our methods (i.e. `} );`), which I don't think would require changes to jscs if we were comfortable with things like `method( arg1, { object: value })`.

Comment (mzgol) [inline build/tasks/build.js]: > But I will point out that I haven't encountered one argument in favor of requiring them that didn't hinge upon style guide automation

I have one - it's just simpler to remember that an opening paren containing something is always followed by a space. Simpler for us, simpler for contributors.

But as long as the tooling enforces the agreed convention, I'm fine whatever we end up with.

Comment (mzgol) [inline build/tasks/build.js]: >
> I'm less bothered by the extra spaces in expression parenthesis as I am by
> the extra space at the end of our methods (i.e. } );), which I don't
> think would require changes to jscs if we were comfortable with things like method(
> arg1, { object: value }).
>

I like those spaces for consistency reasons. If the majority doesn't want
them, I'm fine with this exception (although I prefer not to have too many
exceptions as @mikesherov suggested) but it'd be good to have it checked or
we'll have half of the code in one style & half in the other.

Comment (timmywil) [inline build/tasks/build.js]: For now, I'm okay with leaving it as-is. If we agree to change it, we'll do another PR.
